src/routes/admin/ciraconfig/ciraValidator.ts

```diff
@@ -0,0 +1,158 @@
+/*********************************************************************
+ * Copyright (c) Intel Corporation 2019
+ * SPDX-License-Identifier: Apache-2.0
+ * Author : Madhavi Losetty
+ **********************************************************************/
+import { check } from 'express-validator'
+
+const ipv4 = '^([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])\\.([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])\\.([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])\\.([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])$'
+const ipv6 = '^((?:[0-9A-Fa-f]{1,4}))((?::[0-9A-Fa-f]{1,4}))*::((?:[0-9A-Fa-f]{1,4}))((?::[0-9A-Fa-f]{1,4}))*|((?:[0-9A-Fa-f]{1,4}))((?::[0-9A-Fa-f]{1,4})){7}$'
+const fqdn = '^(?=.{1,254}$)((?=[a-z0-9-]{1,63}\\.)(xn--+)?[a-z0-9]+(-[a-z0-9]+)*\\.)+[a-z]{2,63}$'
+
+export const ciraInsertValidator = (): any => {
+  return [
+    check('payload.configName')
+      .not()
+      .isEmpty()
+      .withMessage('CIRA profile name is required')
+      .matches('^[a-zA-Z0-9$@$!%*#?&-_~^]+$')
+      .withMessage('CIRA profile name accepts letters, numbers, special characters and no spaces'),
+    check('payload.serverAddressFormat')
+      .not()
+      .isEmpty()
+      .withMessage('MPS server address format is required')
+      .isIn([3, 4, 201])
+      .withMessage('MPS server address format should be either 3(IPV4) , 4(IPV6) or 201(FQDN)'),
+    check('payload.mpsServerAddress')
+      .not()
+      .isEmpty()
+      .withMessage('MPS server address is required')
+      .custom((value, { req }) => {
+        if (serverAddressFormatValidator(value, req)) {
+          return true
+        }
+        return false
+      }),
+    check('payload.mpsPort')
+      .not()
+      .isEmpty()
+      .withMessage('MPS port is required')
+      .isInt({ min: 1024, max: 49151 })
```

Comment:
why can't it be below 1024?

Reply:
AMT firmware accepts between 1024 and 49151